recipe/patches/0004-Undo-dmlc-xgboost-9436.patch

```diff
@@ -1,20 +1,20 @@
-From 667a855dfa024ea392e60fa7f352a0781cb7f1e0 Mon Sep 17 00:00:00 2001
-From: Hyunsu Cho <chohyu01@cs.washington.edu>
-Date: Thu, 2 Nov 2023 08:12:43 -0700
-Subject: [PATCH] Undo dmlc/xgboost#9436
+From 3660a4216763d02d48cb1db63f5fee7ad02ad7d4 Mon Sep 17 00:00:00 2001
+From: Hyunsu Cho <phcho@nvidia.com>
```

Comment:
Since the original patch was intended for conda packaging, maybe we can revert it in XGBoost if it's no longer needed?

Reply:
Well there are two things going on

1. Reverting [this RMM workaround]( https://github.com/dmlc/xgboost/blob/411c8466bd383662a448a6a6be585c1b12b8c2e2/CMakeLists.txt#L320-L327 ). AIUI this is needed until RMM fixes issue: https://github.com/rapidsai/rmm/issues/1320
2. Moving `CUDA::cudart_static` into `xgboost_set_cuda_flags`, which is being done in PR: https://github.com/dmlc/xgboost/pull/10633

We can probably cleanup the patches here to make these things clearer (and also make dropping them easier in the future)

Though wanted to wait until the logic here settled (namely we got a green CI) before embarking on that. Otherwise the diff just gets messier and harder to follow while we are still debugging issues

Reply:
Thank you for the explanation, it's really helpful!

Reply:
Happy to help 🙂

Know looking at a diff of a diff is horrendous. So have just been viewing each file directly in its current state

Am encouraged by the work Hyunsu and you have been doing. Think we will be able to start cutting these down in future releases